### PR TITLE
Updated Gloo to v0.0.20200910

### PR DESCRIPTION
**Pull request by stemann**

### CODE REVIEW

**Comment by stemann** [comment]
Windows support currently stuck on `fatal error: BaseTsd.h: No such file or directory`: https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=17983&view=logs&j=5448dbe0-f2cb-5f4c-78ce-09f571b85d5e&t=c83582e6-d39d-5d80-ebce-b51ed759db33&l=290

**Comment by giordano** [comment]
All header files in mingw are lowercase:
```console
sandbox:${WORKSPACE} # qfind /opt/${target}/${target} -iname BaseTsd.h
/opt/x86_64-w64-mingw32/x86_64-w64-mingw32/sys-root/include/basetsd.h
```

File: G/Gloo/bundled/patches/mingw-lowercase-include.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/gloo/types.h b/gloo/types.h
```
Comment (giordano) [inline G/Gloo/bundled/patches/mingw-lowercase-include.patch:1]: It'd be nice to upstream all the patches

Comment (stemann) [reply G/Gloo/bundled/patches/mingw-lowercase-include.patch:1]: Indeed.

**Pull request merged**